app/assets/javascripts/Components/Result/submission_selector.jsx

```diff
@@ -95,6 +95,20 @@ export class SubmissionSelector extends React.Component {
     );
   }
 
+  renderNextRandomUnmarkedSubmissionButton() {
+    if (this.props.role !== "Student") {
+      return (
+        <button
```

Comment:
The submission selector knows how many submissions are still ungraded. So it should be possible to disable this button when there are no other ungraded submissions (other than possible the current submission).